File: Python3/프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    answer = []
    
    for i, j, k in queries:
        temp = arr[i:j+1]
        
        filtered = sorted([x for x in temp if x > k])
        
        if not filtered:
            answer.append(-1)
        else:
            answer.append(filtered[0])
    
    return answer

            
            

File: Swift/프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.swift
import Foundation

func solution(_ arr:[Int], _ queries:[[Int]]) -> [Int] {
    var answer: [Int] = []
    
    for query in queries {
        let i = query[0]
        let j = query[1]
        let k = query[2]
        
        let subArray = arr[i...j]
        
        let filtered = subArray.filter { $0 > k }.sorted()
        
        if filtered.isEmpty {
            answer.append(-1)
        } else {
            answer.append(filtered[0])
        }
    }
    
    return answer
}